core/src/main/java/bisq/core/btc/TxFeeEstimationService.java

```diff
@@ -0,0 +1,159 @@
+
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Util class for getting the estimated tx fee for maker or taker fee tx.
+ */
+@Slf4j
+public class TxFeeEstimationService {
+    public static int TYPICAL_TX_WITH_1_INPUT_SIZE = 260;
+    private final FeeService feeService;
+    private final BtcWalletService btcWalletService;
+    private final Preferences preferences;
+
+    @Inject
+    public TxFeeEstimationService(FeeService feeService,
+                                  BtcWalletService btcWalletService,
+                                  Preferences preferences) {
+
+        this.feeService = feeService;
+        this.btcWalletService = btcWalletService;
+        this.preferences = preferences;
+    }
+
+    public Tuple2<Coin, Integer> getEstimatedFeeAndTxSizeForTaker(Coin fundsNeededForTrade, Coin tradeFee) {
+        return getEstimatedFeeAndTxSize(true,
+                fundsNeededForTrade,
+                tradeFee,
+                feeService,
+                btcWalletService,
+                preferences);
+    }
+
+    public Tuple2<Coin, Integer> getEstimatedFeeAndTxSizeForMaker(Coin reservedFundsForOffer,
+                                                                  Coin tradeFee) {
+        return getEstimatedFeeAndTxSize(false,
+                reservedFundsForOffer,
+                tradeFee,
+                feeService,
+                btcWalletService,
+                preferences);
+    }
+
+    private Tuple2<Coin, Integer> getEstimatedFeeAndTxSize(boolean isTaker,
+                                                           Coin amount,
+                                                           Coin tradeFee,
+                                                           FeeService feeService,
+                                                           BtcWalletService btcWalletService,
+                                                           Preferences preferences) {
+        Coin txFeePerByte = feeService.getTxFeePerByte();
+        // We start with min taker fee size of 260
+        int estimatedTxSize = 260;
+        try {
+            estimatedTxSize = getEstimatedTxSize(List.of(tradeFee, amount), estimatedTxSize, txFeePerByte, btcWalletService);
+        } catch (InsufficientMoneyException e) {
+            if (isTaker) {
+                // if we cannot do the estimation we use the payout tx size
+                estimatedTxSize = 380;
+            }
+            log.info("We cannot do the fee estimation because there are not enough funds in the wallet. This is expected " +
+                    "if the user pays from an external wallet. In that case we use an estimated tx size of {} bytes.", estimatedTxSize);
+        }
+
+        if (!preferences.isPayFeeInBtc()) {
+            // If we pay the fee in BSQ we have one input more which adds about 150 bytes
+            estimatedTxSize += 150;
```

Comment:
Can we avoid magic numbers? https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
Yes we can use a constant. It is just used once so thats why I did not use one and the class is small so easty to overlook all as well there is a comment decribing the value. But no problem to make a constant for it.